src/main/java/htsjdk/samtools/util/IOUtil.java

```diff
@@ -1064,6 +1076,60 @@ public static String slurp(final InputStream is, final Charset charSet) {
     }
 
     /**
+     * Go through the files provided and if they have one of the provided file extensions pass the file into the output
+     * otherwise assume that file is a list of filenames and unfold it into the output.
+     */
+    public static List<Path> unrollPaths(final Collection<Path> inputs, final String... extensions) {
+        if (extensions.length < 1) throw new IllegalArgumentException("Must provide at least one extension.");
+
+        final Stack<Path> stack = new Stack<>();
+        final List<Path> output = new ArrayList<>();
+        stack.addAll(inputs);
+
+        while (!stack.empty()) {
+            final Path p = stack.pop();
+            final String name = p.toUri().toString();
```

Comment:
Why call `.toUri().toString()` here? Seems like `p.toString()` would produce the same result.

Reply:
I don't think it does. it drops the schema.

Reply:
There is no schema for a `Path`. Anyway the code is testing the end of the path, so it doesn't matter what the schema is.

Reply:
sure.